app/middleware/resque_middleware.rb

```diff
@@ -1,19 +1,64 @@
 class ResqueMiddleware
+  # Separate controller instance to access RequestForgeryProtection methods
+  BASE_CONTROLLER = ActionController::Base.new
+
   def initialize(app)
     @app = app
   end
 
+  # CSRF reference: https://medium.com/rubyinside/a-deep-dive-into-csrf-protection-in-rails-19fa0a42c0ef
   def call(env)
+    # Regular execution for non-Resque paths
+    unless env['PATH_INFO'].start_with?("/resque")
```

Comment:
What paths are these? I guess there is not way to mount this middleware class only for certain routes?

Reply:
surprisingly I couldn't find a way with Rails 5 https://stackoverflow.com/questions/12679644/trigger-rack-middleware-on-specific-rails-routes